pkg/store/driver/txn/batch_getter.go

```diff
@@ -132,3 +132,85 @@ func (b *BufferBatchGetter) BatchGet(ctx context.Context, keys []kv.Key) (map[st
 
 	return storageValues, err
 }
+
+// tikvBatchBufferGetter is the BatchBufferGetter struct for tikv
+// In order to directly call NewBufferBatchGetter in client-go
+// We need to implement the interface (transaction.BatchBufferGetter) in client-go for tikvBatchBufferGetter
+type tikvBatchBufferSnapshotGetter struct {
+	tidbMiddleCache    Getter
+	tidbSnapshotBuffer kv.MemBufferSnapshot
```

Comment:
Keeping the naming convention `membufferSnapshot` is better for understandibility I think.